tap_mysql/discover_utils.py

```diff
@@ -37,16 +45,98 @@
     'bigint': 8
 }
 
-FLOAT_TYPES = {'float', 'double'}
+BOOL_TYPES = {'bit'}
+
+JSON_TYPES = {'json'}
+
+FLOAT_TYPES = {'float', 'double', 'decimal'}
 
-DATETIME_TYPES = {'datetime', 'timestamp', 'date'}
+DATETIME_TYPES = {'datetime', 'timestamp', 'time', 'date'}
 
 BINARY_TYPES = {'binary', 'varbinary'}
 
 SPATIAL_TYPES = {'geometry', 'point', 'linestring',
                  'polygon', 'multipoint', 'multilinestring',
                  'multipolygon', 'geometrycollection'}
 
+# A set of all supported column types listed above
+SUPPORTED_COLUMN_TYPES_AGGREGATED = \
+    STRING_TYPES \
+        .union(FLOAT_TYPES) \
+        .union(DATETIME_TYPES) \
+        .union(BINARY_TYPES) \
+        .union(SPATIAL_TYPES) \
+        .union(BOOL_TYPES) \
+        .union(JSON_TYPES) \
+        .union(BYTES_FOR_INTEGER_TYPE.keys())
+
+
+def is_supported_column_type(column_sql_type: str) -> bool:
+    """
+    Checks if the given sql type is supported
+
+    Args:
+        column_sql_type: Column sql data type from the catalog metadata
+            could be in the format {TYPE}(size) or {TYPE}, e.g bigint(100), timestamp ..etc
+
+    Returns: True if column type is supported, False otherwise
+    """
+    try:
+        # handle types where the size is included such as varchar(10)
+        # get the index of the parentheses to use later
+        idx = column_sql_type.index('(')
```

Comment:
if we don't want to do annoying string parsing operations then we can put the short `data_type` into the `properties.json` at discovery time. I think `data_type` is the thing that we are actually looking after, it retrieved directly from MySQL without length and other modifiers. Currently we don't put it into the properties file, but it's possible.

Would be something like this into `discover_util.py` [here](https://github.com/transferwise/pipelinewise-tap-mysql/blob/5a9e043b2581d01229970f8b9debf3452a877515/tap_mysql/discover_utils.py#L242):

```
        mdata = metadata.write(mdata,
                               ('properties', col.column_name),
                               'sql-datatype', col.column_type.lower(),
                               'data_type', col.data_type.lower())
```

Then this function would use `data_type` as the parameter and we would not need to do string trimming. What do you think?

Reply:
That sounds like a great idea, I like it.

Reply:
Updated https://github.com/transferwise/pipelinewise-tap-mysql/pull/61/commits/d1a35c78aaa0ccf13f715223f20f6ab1a0895724